src/main/java/com/cloudbees/jenkins/plugins/awscredentials/AWSCredentialsImpl.java

```diff
@@ -74,24 +74,27 @@
     private final Secret secretKey;
 
     private final String iamRoleArn;
+    private final String iamExternalId;
     private final String iamMfaSerialNumber;
 
     private volatile Integer stsTokenDuration;
 
     // Old data bound constructor. It is maintained to keep binary compatibility with clients that were using it directly.
     public AWSCredentialsImpl(@CheckForNull CredentialsScope scope, @CheckForNull String id,
                               @CheckForNull String accessKey, @CheckForNull String secretKey, @CheckForNull String description) {
-        this(scope, id, accessKey, secretKey, description, null, null);
+        this(scope, id, accessKey, secretKey, description, null, null, null);
     }
 
     @DataBoundConstructor
     public AWSCredentialsImpl(@CheckForNull CredentialsScope scope, @CheckForNull String id,
                               @CheckForNull String accessKey, @CheckForNull String secretKey, @CheckForNull String description,
-                              @CheckForNull String iamRoleArn, @CheckForNull String iamMfaSerialNumber) {
+                              @CheckForNull String iamRoleArn, @CheckForNull String iamMfaSerialNumber,
+                              String iamExternalId) {
```

Comment:
You need to keep the old constructor for compatibility.

Reply:
Brought back old constructor. Please check. Thanks!